core/src/main/java/org/elasticsearch/common/lucene/uid/PerThreadIDAndVersionLookup.java

```diff
@@ -51,49 +51,52 @@
     private final TermsEnum termsEnum;
     /** _version data */
     private final NumericDocValues versions;
+
     /** Reused for iteration (when the term exists) */
     private PostingsEnum docsEnum;
 
+    private final Object readerKey;
```

Comment:
this reader key is only used for asserts. can we make sure it's null if asserts are not enabled?